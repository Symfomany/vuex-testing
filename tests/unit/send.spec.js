import { shallowMount, mount } from "@vue/test-utils"
import Send from "@/components/Send.vue";

import Vuetify from "vuetify";
import Vue from "vue";
import { VIcon } from "vuetify/lib";

Vue.use(Vuetify);


describe("Send", () => {
  it("reveals a notification when submitted", () => {
        Vue.use(Vuetify);

      const wrapper = shallowMount(Send)
  expect(wrapper.find(VIcon).exists()).toBe(true); // Passes

    // expect(wrapper.find(".message").text())
    //   .toBe("Thank you for your submission, alice.")
  //   wrapper.find("[data-username]").setValue("alice")
  //   // console.log( wrapper.find("button").attributes());
  //   wrapper.find("form").trigger("submit.prevent")
  //   // console.log(wrapper.html());
  //   expect(wrapper.element).toMatchSnapshot();

  //   expect(wrapper.find("#mess").text())
  //     .toBe("Thank you for your submission, alice.")
  })
})