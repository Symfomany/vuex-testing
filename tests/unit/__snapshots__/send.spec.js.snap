// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Send reveals a notification when submitted 1`] = `
<div>
  <form
    class="v-form"
    novalidate="novalidate"
  >
    <div
      class="container grid-list-xl"
    >
      <div
        class="layout wrap"
      >
        <div
          class="flex xs12 md4"
        >
          <div
            class="v-input theme--light v-text-field"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    class="v-label theme--light"
                    for="input-3"
                    style="right: 0px; position: absolute;"
                  >
                    Username
                  </label>
                  <input
                    data-username=""
                    id="input-3"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <span
                    class="v-messages__wrapper"
                    name="message-transition"
                    tag="div"
                  />
                </div>
              </div>
            </div>
          </div>
           
          <button
            class="v-btn v-btn--contained theme--light v-size--default primary"
            type="submit"
          >
            <span
              class="v-btn__content"
            >
              Envoyer
            </span>
          </button>
        </div>
      </div>
    </div>
  </form>
   
  <!---->
</div>
`;
